android/app/src/main/java/edu/berkeley/eecs/e_mission/data_sync/ConfirmTripsAdapter.java

```diff
@@ -62,118 +105,302 @@ public ConfirmTripsAdapter(Context context, boolean autoInitialize) {
+				 * This is probably not a huge issue initially for most people, but some cases that
+				 * I can think of are:
+				 * - two people sharing the same account like Tom and me
+				 * - people using a different device for the data collection and a different one
+				 * (google watch) for the classification.
+				 *
+				 * The database on the server is the source of truth, so we can always just synchronize to it.
+				 * So let's clear our local cached copy before pulling data from the server.
+				 */
+
+				if (Service.equals("Ours")) {
+					dbHelper.clear();
+				}
+
+				JSONArray unclassifiedSections = CommunicationHelper.getUnclassifiedSections(
+						cachedContext, userToken);
+				statsHelper.storeMeasurement(cachedContext.getString(R.string.sync_pull_list_size),
+						String.valueOf(unclassifiedSections.length()), syncTs);
+				// save the unclassified sections to the database
+				if (Service.equals("Ours")) {
+					dbHelper.storeNewUnclassifiedTrips(convertJSONToList(unclassifiedSections));
+				} else if (Service.equals("CouchBase")) {
+					Document document = database.createDocument();
+					Map<String, Object> properties = new HashMap<String, Object>();
+					properties.put("unclassifiedSections", convertJSONToList(unclassifiedSections));
+					try {
+						document.putProperties(properties);
+					} catch (CouchbaseLiteException e) {
+						e.printStackTrace();
+					}
+				}
+				// We should be able to figure out which sections are already in the notification bar and filter those out
+				// But let's do something lame and easy for now
+				// TODO: Optimize this later
+				// Also, since we are already using a database, we may want to consider
+				// using a content provider, since that is the recommended solution
+				// TODO: Simplify this later, once we are sure that the list stuff works
+				generateNotifications(unclassifiedSections);
+
+				// Now, we push the stats and clear it
+				// Note that the database ensures that we have a blank document if there are no stats
+				// by skipping the metadata in that case.
+				JSONObject freshStats = statsHelper.getMeasurements();
+				if (freshStats.length() > 0) {
+					CommunicationHelper.pushStats(cachedContext, userToken, freshStats);
+					statsHelper.clear();
+				}
+				statsHelper.storeMeasurement(cachedContext.getString(R.string.sync_duration),
+						String.valueOf(System.currentTimeMillis() - msTime), syncTs);
+			} catch (IOException e) {
+				// TODO Auto-generated catch block
+				e.printStackTrace();
+			} catch (JSONException e) {
+				// TODO Auto-generated catch block
+				e.printStackTrace();
+			}
+
+			try {
+				updateResultsSummary(userToken);
+			} catch (MalformedURLException e) {
+				e.printStackTrace();
+			} catch (IOException e) {
+				e.printStackTrace();
+			}
+		//} else if (Service.equals("Azure")) {
+			///
+		if (Service.equals("CouchBase")) {
+			// Couchbase
+			Replication push = database.createPushReplication(SYNC_URL);
```

Comment:
In particular, given the previous comment, I am not sure if this is doing anything.